src/Microsoft.TestPlatform.Utilities/InferRunSettingsHelper.cs

```diff
@@ -675,15 +678,30 @@ private static bool IsPlatformIncompatible(Architecture sourcePlatform, Architec
         }
 
         /// <summary>
-        /// Returns true if source FrameworkVersion is incompatible with target FrameworkVersion.
+        /// Returns true if source Framework or version is incompatible with target Framework or version.
         /// </summary>
-        private static bool IsFrameworkIncompatible(Framework sourceFramework, Framework targetFramework)
+        public static bool IsFrameworkIncompatible(IEnumerable<Framework> sourceFrameworks, Framework targetFramework, bool versionCheckRequired = true)
         {
-            if (sourceFramework.Name.Equals(Framework.DefaultFramework.Name, StringComparison.OrdinalIgnoreCase))
+            bool isFrameworkInCompatible = false;
+
+            foreach (var actualFramework in sourceFrameworks)
             {
-                return false;
+                if (versionCheckRequired)
+                {
+                    isFrameworkInCompatible = isFrameworkInCompatible || !actualFramework.Name.Equals(targetFramework.Name, StringComparison.OrdinalIgnoreCase);
+                }
+
+                string sourceFrameworkName = actualFramework.Name.Split(',')[0];
+                string targetFrameworkName = targetFramework.Name.Split(',')[0];
+
+                if (sourceFrameworkName.Equals(DotNetFrameworkString, StringComparison.OrdinalIgnoreCase) && targetFrameworkName.Equals(DotNetCoreString, StringComparison.OrdinalIgnoreCase) ||
```

Comment:
>DotNetFrameworkString [](start = 47, length = 21)

Lets keep the framework comparison logic same as above, we shouldn't have to compare them with Const strings, & mark them as incompatible.